Client/Application/AdjustIntegration.swift

```diff
@@ -0,0 +1,129 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Foundation
+import Shared
+
+private let AdjustIntegrationErrorDomain = "org.mozilla.ios.Firefox.AdjustIntegrationErrorDomain"
+
+private let AdjustAttributionFileName = "AdjustAttribution.json"
+
+private let AdjustAppTokenKey = "ADJUST_APP_TOKEN"
+private let AdjustEnvironmentKey = "ADJUST_ENVIRONMENT"
+
+private let AdjustSandboxEnvironment = "sandbox"
+private let AdjustProductionEnvironment = "production"
+
+private enum AdjustEnvironment: String {
+    case Sandbox = "sandbox"
+    case Production = "production"
+}
+
+private struct AdjustSettings {
+    var appToken: String
+    var environment: AdjustEnvironment
+}
+
+/// Simple (singleton) object to contain all code related to Adjust. The idea is to capture all logic
+/// here so that we have one single place where we can see what Adjust is doing. Ideally you only call
+/// functions in this object from other parts of the application.
+
+class AdjustIntegration: NSObject {
+    static let sharedInstance = AdjustIntegration()
+
+    /// Return an ADJConfig object if Adjust has been enabled. It is determined from the values in
+    /// the Info.plist file if Adjust should be enabled, and if so, what its application token and
+    /// environment are. If those keys are either missing or empty in the Info.plist then it is
+    /// assumed that Adjust is not enabled for this build.
+
+    private func getConfig() -> ADJConfig? {
+        guard let settings = getSettings() else {
+            return nil
+        }
+
+        let config = ADJConfig(appToken: settings.appToken, environment: settings.environment.rawValue)
+        if settings.environment == .Sandbox {
+            config.logLevel = ADJLogLevelDebug
+        }
+        config.delegate = self
+        return config
+    }
+
+    /// Returns the Adjust settings from our Info.plist. If the settings are missing or invalid, such as an unknown
+    /// environment, then it will return nil.
+
+    private func getSettings() -> AdjustSettings? {
+        let bundle = NSBundle.mainBundle()
+        guard let adjustAppToken = bundle.objectForInfoDictionaryKey(AdjustAppTokenKey) as? String, adjustEnvironment = bundle.objectForInfoDictionaryKey(AdjustEnvironmentKey) as? String else {
+            return nil
+        }
+        if adjustAppToken.isEmpty || adjustEnvironment.isEmpty {
```

Comment:
For consistency, maybe make this a `guard adjustAppToken.isEmpty || adjustEnvironment.isEmpty else { ... }`?